### DISCUSSION TITLE
Simplify your full-stack applications with XState 📒

### DISCUSSION THREAD

**Discussion by danigo99** [statelyai/xstate#3448] [Show and tell]
Hey everyone, I’ve just posted an article about XState which I thought was worth sharing here. I used XState on an earlier project to manage a complex flow of medication requests through a NestJS API, it improved our developer experience, and provided an intuitive bridge for non-technical stakeholders. In this article I talk about some of its best features as well as how you can start using it in your backend applications right now!

Feel free to ask any questions or leave feedback, and hope you enjoy!

https://blog.theodo.com/2022/07/simplify-your-applications-with-xstate/

**Comment by davidkpiano** [comment]
Fantastic article, thanks for sharing it!

**Reply by danigo99** [reply]
Thanks a lot David! And thank you for this great tool :)

**Comment by Jfsslemos** [comment]
Hi, thanks for your post. I'm having a hard time to understand how the structure should look like

**Reply by davidkpiano** [reply]
Hey @Jfsslemos, what are you trying to accomplish? I can help.

**Reply by Jfsslemos** [reply]
I could not understand what should be the application's final structure. What should go in the services,  what should go in the controller etc

**Reply by Jfsslemos** [reply]
I'm talking focused on the nestjs and xstate integration. I'm pretty new in both technologies
